### PR TITLE
Electrolyzer Improvements part 1

### PR DESCRIPTION
**Pull request by 70000hp**
Changes:
Halfed recipe time for water/heavy water electrolysis 
Added ore slop electrolysis recipe
Added electrolyzer overdrive support
Rebalanced the power savings upgrade
Speed and overdrive upgrades now make the electrolyzer pour out faster
Added adjustable recipe times to both solid and liquid modes

### CODE REVIEW

**Comment by TreeVirus** [comment]
Bath.

**Comment by signed-firmware** [comment]
Bath.

**Comment by VettiusCattusXVII** [comment]
Supercalifragilisticexpialidocious.

**Comment by SoB1nary** [comment]
Bath.

File: src/main/java/com/hbm/tileentity/machine/TileEntityElectrolyser.java [hunk 1]
```diff
@@ -501,24 +520,30 @@ public boolean hasPermission(EntityPlayer player) {
...
+		if(type == UpgradeType.OVERDRIVE) {
+			info.add((BobMathUtil.getBlink() ? EnumChatFormatting.RED : EnumChatFormatting.DARK_GRAY) + "YES");
 		}
 	}
 
 	@Override
 	public int getMaxLevel(UpgradeType type) {
 		if(type == UpgradeType.SPEED) return 3;
 		if(type == UpgradeType.POWER) return 3;
+		if(type == UpgradeType.OVERDRIVE) return 2;
```
Comment (HbmMods) [inline src/main/java/com/hbm/tileentity/machine/TileEntityElectrolyser.java]: the cycle count implies that the max level is 3

File: src/main/java/com/hbm/inventory/recipes/ElectrolyserFluidRecipes.java [hunk 2]
```diff
@@ -24,9 +26,10 @@ public class ElectrolyserFluidRecipes extends SerializableRecipe {
 
 	@Override
 	public void registerDefaults() {
-		recipes.put(Fluids.WATER, new ElectrolysisRecipe(2_000, new FluidStack(Fluids.HYDROGEN, 200), new FluidStack(Fluids.OXYGEN, 200)));
-		recipes.put(Fluids.HEAVYWATER, new ElectrolysisRecipe(2_000, new FluidStack(Fluids.DEUTERIUM, 200), new FluidStack(Fluids.OXYGEN, 200)));
+		recipes.put(Fluids.WATER, new ElectrolysisRecipe(2_000, new FluidStack(Fluids.HYDROGEN, 200), new FluidStack(Fluids.OXYGEN, 200),10));
```
Comment (HbmMods) [inline src/main/java/com/hbm/inventory/recipes/ElectrolyserFluidRecipes.java:29]: we had a more than lengthy argument about this when the recipes were rebalanced, i don't see half process time as justified

File: src/main/java/com/hbm/tileentity/machine/TileEntityElectrolyser.java (hunk 1 above)
Comment (70000hp) [reply src/main/java/com/hbm/tileentity/machine/TileEntityElectrolyser.java]: It's the same impl as acidizer code
Now why is the max level of it 2? No idea

File: src/main/java/com/hbm/inventory/recipes/ElectrolyserFluidRecipes.java (hunk 2 above)
Comment (70000hp) [reply src/main/java/com/hbm/inventory/recipes/ElectrolyserFluidRecipes.java:29]: I tested it extensively, and this allows the electrolyzer to roughly catch up to the chemfac, and be better than it at electrolysis for the cost

File: src/main/java/com/hbm/tileentity/machine/TileEntityElectrolyser.java (hunk 1 above)
Comment (70000hp) [reply src/main/java/com/hbm/tileentity/machine/TileEntityElectrolyser.java]: Fixed

**Pull request merged**